src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -41,6 +42,13 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deletePerson(personToDelete);
+
+        // TODO: check current tab and only switch if user is on PersonDetailsTab
+        if (model.getDetailedPerson().contains(personToDelete)) {
```

Comment:
Will the `model.getDetailedPerson()` be emptied out when switching to other panels?

Reply:
i.e. will the `list` be emptied if the app is not showing the detailedPerson view?

Reply:
Nope. All this does is check if the detailedPerson is the personToDelete and if so, switches the tab to the UpcomingEvents tab.

The limitation of such an implementation is that if they are not on the detailedPersonTab, (i.e. streaks tab) the details panel will still switch to upcoming events tab.

Reply:
Ok got it